evm-types.md

```diff
@@ -253,14 +252,10 @@ The helper `powmod` is a totalization of the operator `_^%Int__` (which comes wi
     syntax Int ::= Int "/sWord" Int [function]
                  | Int "%sWord" Int [function]
  // ------------------------------------------
-    rule [divSWord]: W0 /sWord W1 => #sgnInterp(sgn(W0) *Int sgn(W1) , abs(W0) /Word abs(W1))
-    rule [modSWord]: W0 %sWord W1 => #sgnInterp(sgn(W0)              , abs(W0) %Word abs(W1))
-
-    syntax Int ::= #sgnInterp ( Int , Int ) [function, functional]
- // --------------------------------------------------------------
-    rule #sgnInterp( W0 ,  _ ) => 0          requires W0 ==Int 0
-    rule #sgnInterp( W0 , W1 ) => W1         requires W0 >Int 0
-    rule #sgnInterp( W0 , W1 ) => 0 -Word W1 requires W0 <Int 0
+    rule [divSWord.same]: W0 /sWord W1 =>          abs(W0) /Word abs(W1)  requires sgn(W0) *Int sgn(W1) ==Int  1
+    rule [divSWord.diff]: W0 /sWord W1 => 0 -Word (abs(W0) /Word abs(W1)) requires sgn(W0) *Int sgn(W1) ==Int -1
+    rule [modSWord.pos]:  W0 %sWord W1 =>          abs(W0) %Word abs(W1)  requires sgn(W0) ==Int  1
+    rule [modSWord.neg]:  W0 %sWord W1 => 0 -Word (abs(W0) %Word abs(W1)) requires sgn(W0) ==Int -1
```

Comment:
Don't you need the rule for the case of `sgn(W0) *Int sgn(W1) == 0` and `sgn(W0) == 0`?

Although it is an undefined case, you had a rule for that before, but not now.  Just to make sure you know this.

Reply:
Yeah I'm aware of that change, but it seems OK because both `_/sWord_` and `_%sWord_` were not defined as `functional` before, and they still aren't defined as such.

So I figured it was OK to just define the sensible cases here, because the proofs still pass. I can add in the other cases (going to 0) later if I find that it's needed to make those `functional`.

Reply:
In the case of `_/sWord_` and `_%sWord_` we know that the semantics should never feed it anything that is not an EVM Word, so it should always be defined at runtime.